cdap-gateway/src/main/java/co/cask/cdap/gateway/discovery/UserServiceEndpointStrategy.java

```diff
@@ -39,48 +39,71 @@
   private final ProgramId serviceId;
   private final String version;
   private final Random random;
+  private final RouteFallbackStrategy fallbackStrategy;
 
   public UserServiceEndpointStrategy(ServiceDiscovered serviceDiscovered, RouteStore routeStore, ProgramId serviceId,
-                                     @Nullable String version) {
+                                     String fallbackStrategy, @Nullable String version) {
```

Comment:
Better take a `RouteFallbackStrategy`. Turning a string into that enum is not the responsibility of this class.